implementation/src/main/java/io/smallrye/jwt/auth/AbstractBearerTokenExtractor.java

```diff
@@ -39,9 +39,11 @@
     private static final Logger LOGGER = Logger.getLogger(AbstractBearerTokenExtractor.class);
 
     private final JWTAuthContextInfo authContextInfo;
+    private final JWTParser jwtParser;
 
-    protected AbstractBearerTokenExtractor(JWTAuthContextInfo authContextInfo) {
+    protected AbstractBearerTokenExtractor(JWTAuthContextInfo authContextInfo, JWTParser jwtParser) {
         this.authContextInfo = authContextInfo;
+        this.jwtParser = jwtParser;
```

Comment:
@darranl Can this parser be injected ? Otherwise we'd need to change Thorntail and Quarkus code a bit to pass it around as well, so the `smallrye-jwt` version will likely have to changed to 3.0.0 - not a problem in itself, but please check

Reply:
@sberyozkin I will have a look, at the moment I just injected where we already have injection occurring, I think the final place it is used wasn't a CDI bean.

Reply:
@darranl I've looked at Quarkus/Thorntail code, not sure we can avoid passing `JWTParser` to the token extractors, but I'm also not sure either now `JWTParser` needs to be passed to them :-), it is only used indirectly [here](https://github.com/smallrye/smallrye-jwt/blob/master/implementation/src/main/java/io/smallrye/jwt/auth/jaxrs/JWTAuthenticationFilter.java#L64), which is where it is actually have to be used directly because the extractor is not really about the validation. So please make that `validate` method in the abstract extractor code deprected and use the parser directly in the JAX-RS filter which is where is it already injected to in your PR and it should do it

Thanks

Reply:
@darranl In fact I'd just remove that `valdate` method since the only consumer of that method is this JAX-RS filter AFAIK. But I'm OK if we stay on the safer side and only deprecate it. CC @MikeEdgar Hi Mike, FYI

Reply:
@sberyozkin - +1 to removing `validate`. I believe it was originally copied from Thorntail and made public so that it was callable from the JAX-RS filter and the HTTP auth mechanism class.